sql/_13_issues/_22_1h/cases/cbrd_24149.sql

```diff
@@ -0,0 +1,251 @@
+
+-- in operation
+select /*+ recompile */ var_col from tbl1 where var_col in (cast('a  ' as varchar),cast('ab     ' as varchar)) using index none;
+select /*+ recompile */ var_col from tbl1 where var_col in (cast('a  ' as varchar),cast('ab     ' as varchar)) using index char_idx;
+select /*+ recompile */ var_col from tbl1 where var_col in (cast('a  ' as varchar),cast('ab     ' as varchar))  using index midx_idx2;
+select /*+ recompile index_ss */ var_col from tbl1 where var_col in (cast('a  ' as varchar),cast('ab     ' as varchar)) using index midx_idx1;
+
+select /*+ recompile */ var_col from tbl1 where var_col in (var_col,var_col);
+select /*+ recompile */ var_col from tbl1 where (var_col,var_col) in ((var_col,var_col),(var_col,var_col));
+
+-- string function (decode)
+select /*+ recompile */ var_col, decode(var_col,cast('a' as varchar),1,cast('a  b' as varchar(20)),2,cast('ab' as varchar(20)),3,4) from tbl1;
+
+-- prepare statement
+prepare v_stmt from 'select var_col as "prepare_v" from tbl1 where var_col = cast(? as varchar(20))';
+execute v_stmt using 'a';
+execute v_stmt using 'a     ';
+drop prepare v_stmt;
+
+
+
+-- reduce duplicate
+
+--eliminate_duplicated_keys
+select /*+ recompile */ var_col from tbl1 where var_col in ('a  ','a ','ab     ') using index none;
+select /*+ recompile */ var_col from tbl1 where var_col in ('a  ','a ','ab     ') using index midx_idx1;
+select /*+ recompile */ var_col from tbl1 where var_col in ('a  ','a ','ab     ') using index var_idx;
+
+select /*+ recompile */ var_col from tbl1 where char_col in ('a  ','a ','ab     ') using index none;
+select /*+ recompile */ var_col from tbl1 where char_col in ('a  ','a ','ab     ') using index midx_idx2;
+select /*+ recompile */ var_col from tbl1 where char_col in ('a  ','a ','ab     ') using index char_idx;
+
+--merge_key_ranges, qo_reduce_comp_pair_terms
+select /*+ recompile */ var_col from tbl1 where var_col > 'a' and var_col > 'a ' and var_col <= 'a     ' using index none;
+select /*+ recompile */ var_col from tbl1 where var_col > 'a' and var_col > 'a ' and var_col <= 'a     ' using index midx_idx1;
+select /*+ recompile */ var_col from tbl1 where var_col > 'a' and var_col > 'a ' and var_col <= 'a     ' using index var_idx;
+
+select /*+ recompile */ char_col from tbl1 where char_col> 'a' and char_col> 'a ' and char_col<= 'a     ' using index none;
+select /*+ recompile */ char_col from tbl1 where char_col> 'a' and char_col> 'a ' and char_col<= 'a     ' using index midx_idx1;
+select /*+ recompile */ char_col from tbl1 where char_col> 'a' and char_col> 'a ' and char_col<= 'a     ' using index var_idx;
+
+select /*+ recompile */ var_col from tbl1 where var_col > 'a' and var_col > 'a ' using index none;
+select /*+ recompile */ var_col from tbl1 where var_col > 'a' and var_col > 'a ' using index midx_idx1;
+select /*+ recompile */ var_col from tbl1 where var_col > 'a' and var_col > 'a ' using index var_idx;
+
+
+
+
+-- create index error
```

Comment:
the comment should be modified not to produce the error.

Reply:
fix it